analytics/src/test/java/com/segment/analytics/internal/AnalyticsClientTest.java

```diff
@@ -76,13 +79,48 @@ AnalyticsClient newClient() {
   }
 
   @Test
-  public void enqueueAddsToQueue(MessageBuilderTest builder) throws InterruptedException {
+  public void enqueueAddsToQueue(MessageBuilderTest builder) {
     AnalyticsClient client = newClient();
 
     Message message = builder.get().userId("prateek").build();
     client.enqueue(message);
 
-    verify(messageQueue).put(message);
+    verify(messageQueue).offer(message);
+  }
+
+  @Test
+  public void enqueueDropsMessageWhenQueueFull(MessageBuilderTest builder) {
+    BlockingQueue messageQueue = new LinkedBlockingQueue<Message>(10);
+    // simulate indefinite blocking on submission to network executor
+    doAnswer(new Answer() {
+      @Override
+      public Object answer(InvocationOnMock invocation) throws Throwable {
+        Thread.sleep(3600 * 1000);
+        return null;
+      }
+    }).when(networkExecutor).submit(any(Runnable.class));
+
+    AnalyticsClient analyticsClient = new AnalyticsClient(
+      messageQueue,
+      segmentService,
+      10,
+      TimeUnit.HOURS.toMillis(1),
+      log,
+      threadFactory,
+      networkExecutor,
+      Collections.singletonList(callback));
```

Comment:
You mind using the `AnalyticsClient.create` to add up to coverage? Since we're changing the body of .create too